ral-facilities/operationsgateway
# DSEGOG-69 DSEGOG-89 attempt to fix the date-time pickers

## Description
This attempts to improve the existing date-time pickers functionality. 

- I've harmonised the way we capture and update date parameters for the search query with the way DataGateway does it. This was originally an attempt to fix the behaviour of the DateTimePickers being partially updated in the picker menu and not updating the query but this didn't solve the actual problem
- Fixing a bug where pressing the Enter key reloaded the page. This was because of the surrounding `form` tag, which I've replaced with a `Grid` to get the same effect.
- Adding CLEAR, CANCEL and OK buttons. The picker NEARLY works perfectly, but I can't achieve the same functionality with the OK button as in DataGateway. It seems the entirety of the field needs to be completed in the picker before the value is accepted. This is not true if the user inputs the value manually. 

So essentially - the OK button is useless in a DateTimePicker! Because, regardless, it doesn't make the picker accept what is currently filled in the input field. These are issues that [others have noticed](https://github.com/mui/mui-x/issues/4599) and developers [deliberately want this behaviour to continue for various reasons](https://github.com/mui/mui-x/issues/6161).

As a result, I've had to stick with the existing combined usage of the `onChange` and `onAccept` functions to achieve a decent amount of the functionality we want between the picker and the text input. So I'm eager to get some input from Louise on this one. We can either:

- stick with the existing functionality and I remove the pointless OK button
- we use the functionality provided by the developers where the `onChange` method is used to update the search parameters every time it calls (this means the table will update for each click of the user selecting date, hour, minute and second)
- some other option I haven't thought of?

Bottom line - it seems DatePickers behave quite differently from DateTimePickers.

## Testing instructions
Check that pressing Enter no longer reloads the page. Check the CLEAR and CANCEL buttons work as intended. Suggest anything else that might be improved.


## Agile board tracking
Closes DSEGOG-69, DSEGOG-89
